ext/sqlite3/tests/sqlite3_38_extended_error_2.phpt

```diff
@@ -0,0 +1,36 @@
+--TEST--
+SQLite3 extended error code Function
+--EXTENSIONS--
+sqlite3
+--FILE--
+<?php
+
+require_once(__DIR__ . '/new_db.inc');
+
+$db->query("CREATE TABLE dog ( id INTEGER PRIMARY KEY, name TEXT, annoying INTEGER )");
+
+echo "Inserting first time which should succeed" . PHP_EOL;
```

Comment:
I notice that these tests only test INSERT, so the "call _reset if _DONE" fix I've done covers it, but it's late and I'm drawing a blank on a SELECT which fails only on the second time you run it (short of using a file database and deleting a table from a second instance, maybe), but it feels like there could be something obvious that just isn't coming to mind.

If it is an issue, could we just run sqlite3_reset whenever lastErrorCode is called? It feels like a bit of a strange side effect to reset the fetch position but I can't really imagine a situation where you'd be checking error codes inside the fetchArray while loop.